pubpub/platform
# Add `isTitle` to pubtype-pubfield relationships

## Issue(s) Resolved

Closes https://github.com/pubpub/platform/issues/735

## High-level Explanation of PR
* Adds a new `isTitle` column to the join table between PubType and PubFields.
* Users must specify a PubField as the "name" for a pubtype when creating a pub type

## Test Plan
* Migrate your local database. This should run the migration, which designates a field as the "title" for each pub type. Make sure all of your pub types now have a pub field designated as 'isTitle'!
* Create a new pub type. You should be able to specify a field as the title
* Edit a pub type. You should be able to change which field is the title

## Screenshots (if applicable)
Pub types
![image](https://github.com/user-attachments/assets/40372b6f-3db2-48de-956f-cb6a516aff7b)

Editing pub type
![image](https://github.com/user-attachments/assets/92334eda-5bdd-43a7-abb1-721eb340477a)
* Only two options are available to be chosen as a title because those are the only string/number/richtext fields
* The delete button is disabled if the option is the current title

Creating a pub type
![image](https://github.com/user-attachments/assets/9a49c657-fa4f-4933-9c62-d81b49855b8b)

If no field is chosen as the name field
![image](https://github.com/user-attachments/assets/21ff0308-0a0b-4fe5-a469-ed94ffd540e0)



## Notes

Nothing uses these name fields yet. `PubTitle` would be a good candidate, though I had some trouble with the prisma query there, so figured I'd put this PR up first since it's already fairly chunky
